daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1033,15 +1033,9 @@ private[lf] object SBuiltin {
       onLedger.cachedContracts.get(coid) match {
         case Some(cached) =>
           if (cached.templateId != templateId) {
-            if (onLedger.ptx.localContracts.contains(coid)) {
```

Comment:
Maybe we should add a test that would detect this?

Reply:
I am adding a test for that in `com.daml.lf.speedy.EvaluationOrderTest`

Reply:
Thanks :)